plugins/torrent/deluge_

```diff
@@ -0,0 +1,399 @@
+   host 127.0.0.1
+   port 58846
+   no username
+   no password
+
+You can change these settings in "plugin-conf.d/munin-node" :
+    [deluge_*]
+   user <user_with_access_to_deluge>
+   env.HOME <path_to_deluge_user_home>
+   env.host 127.0.0.1
+   env.port 58846
+   env.username user
+   env.password pass
+
+By default, deluge configuration files will be searched under $XDG_CONFIG_HOME,
+which is by default set to $HOME/.config
+Setting env.HOME allows this default to work. However, you can also explicitly
+set the env.XDG_CONFIG_HOME if needed.
+
+=head1 INTERPRETATION
+
+=head2 connections
+
+In the "connections" mode, this plugin shows a graph with the number of
+connections
+
+=head2 bandwidth
+
+In the "bandwidth" mode, this plugin show graphs for the download and upload
+bandwidths.
+Each of them has "payload" and "overhead" value.
+ - with positive values : the upload values
+ - with negative values : the download values
+
+=head2 states
+
+In the "states" mode, this plugin shows the number of torrents in each state :
+Downloading, Seeding, Paused, Error, Queued, Checking, Other
+
+=head1 MAGIC MARKERS
+
+  #%# family=auto
+  #%# capabilities=autoconf suggest
+
+=head1 VERSION
+
+1.0.0
+
+=head1 AUTHOR
+
+Neraud (https://github.com/Neraud)
+
+=head1 LICENSE
+
+GPLv2
+
+=cut"""
+
+
+import logging
+import os
+import string
+import sys
+
+try:
+    from deluge.log import setupLogger
+    from deluge.ui.client import client
+    from twisted.internet import reactor, defer
+    setupLogger()
+except (ImportError, NameError):
+    successful_import = False
+else:
+    successful_import = True
+
+
+plugin_version = "1.0.0"
+
+log = logging.getLogger("delugeStats")
+log.setLevel(logging.WARNING)
+
+conf = {
+    'host':       os.getenv('host', '127.0.0.1'),
+    'port':       int(os.getenv('port', '58846')),
+    'username':   os.getenv('username', ''),
+    'password':   os.getenv('password', '')
+}
+
+names_for_munin = {
+    'numConnections':        'numConnections',
+    'payloadUploadRate':     'payloadUploadRate',
+    'overheadUploadRate':    'overheadUploadRate',
+    'payloadDownloadRate':   'payloadDownloadRate',
+    'overheadDownloadRate':  'overheadDownloadRate',
+    'state.Seeding':         'seeding',
+    'state.Downloading':     'downloading',
+    'state.Paused':          'paused',
+    'state.Error':           'error',
+    'state.Queued':          'queued',
+    'state.Checking':        'checking',
+    'state.Other':           'other'
+}
+
+torrent_states = ["Downloading",
+                  "Seeding",
+                  "Paused",
+                  "Error",
+                  "Queued",
+                  "Checking",
+                  "Other"]
+
+modes = ["bandwidth", "connections", "states"]
+
+
+class StatClient:
+
+    def __init__(self, conf, mode):
+        self.conf = conf
+        self.mode = mode
+        self.connected = False
+
+    def end_session(self, msg):
+        log.debug("end_session : {0}".format(msg))
```

Comment:
Maybe you want to use lazy evaluation for log messages (see [logging documentation](https://docs.python.org/3/library/logging.html#logging.debug)):
```python
log.debug("end session: %s", msg)
```
(thus the format strings would only evaluated if the log level is met)